extensions/objects/models/objects.py

```diff
@@ -1045,3 +1045,76 @@ class DragAndDropPositiveInt(BaseObject):
     default_value = 1
 
     SCHEMA = PositiveInt.SCHEMA
+
+
+class AlgebraicExpression(BaseObject):
+    """Class for algebraic expressions. Stores a unicode string representing a
+    valid algebraic expression.
+    The expression should be in the ASCIIMath format.
+    More info: http://asciimath.org/
+    """
+
+    description = 'A unicode string for an algebraic expression.'
+    default_value = 'a'
+
+    SCHEMA = {
+        'type': 'unicode',
+        'validators': [{
+            'id': 'is_valid_asciimath_expression',
+            'algebraic': True
+        }]
+    }
+
+
+class SetOfPlaceholders(BaseObject):
+    """Class for set of placeholders. Placeholders are latin letters that are
+    used in Math Interactions.
+    A placeholder could be an english alphabet (uppercase/lowercase) or a greek
+    letter represented as a single word. Valid greek letters are present in the
+    GREEK_LETTERS constant in schema_utils.py.
+    """
+
+    description = 'A set (a list with unique elements) of placeholders.'
+    default_value = []
+
+    SCHEMA = {
+        'type': 'list',
+        'items': UnicodeString.SCHEMA,
+        'validators': [{
+            'id': 'is_uniquified'
+        }, {
+            'id': 'contains_valid_placeholders'
```

Comment:
> I've updated the validator name along with the docstring of the validator function. I don't see why we need to use "choices"? The validation is quite straightforward so it doesn't make sense listing down all the possible letters instead.

Because you're introducing an entirely new custom function, which is a broader change than using an existing function that is known to work well and that has been battle-tested. So, if you have existing infrastructure that already does the job, use that, rather than creating something completely custom that is relevant only to your specific interaction(s).

(You can still store that list in a named constant so that it doesn't need to be duplicated each time.)

Reply:
Understood. I've made the changes.